hw4/slavrom.js

```diff
@@ -0,0 +1,115 @@
+// 1
+
+let range = {
+  from: 1,
+  to: 10
+}
+
+range[Symbol.iterator] = function() {
+  return {
+    i: 1,
+    next() {
+      if (this.i <= 10) {
+        return { value: this.i++, done: false };
+      }
+      return { value: undefined, done: true };
+    }
+  };
+}
+
+for (let num of range) {
+  console.log(num)
+}
+
+// or
+
+range[Symbol.iterator] = function* () {
+  let index = 1;
+  while (index <= 10) {
+    yield index++;
+  }
+}
+
+for (let num of range) {
+  console.log(num)
+}
+
+// 2
+
+oddNum = (num) => {
+  num = [...num];
+  let odd = [];
+  for (let i = 0; i < num.length; i++) {
+    if (num[i] % 2) {
+      odd.push(num[i]);
+    }
+  }
+  return console.log(odd);
+}
+
+oddNum(range)
+
+// 3 
+
+evenNum = (num) => {
+  num = [...num];
+  return console.log(num.filter(function(val) {return val % 2 === 0}));
+}
+
+evenNum(range);
+
+// 4
+
+argumentsLogger = (...arguments) => {
```

Comment:
this should be not an arrow function but regular one with empty brackets. just like it shown in the task description.